### DISCUSSION TITLE
Is there any API to detect if a Rect is inside/on boundary/outside of another Rect?

### DISCUSSION THREAD

**Discussion by rlin-flex** [georust/geo#1280] [Q&A]
Wrong message

**Comment by michaelkirk** [comment]
> Is there any API to detect if a `Rect` is inside/on boundary/outside of another `Rect`?

For checks like these, your best bet is the [Topology](https://docs.rs/geo/latest/geo/index.html#topology) section. In particular, the `Relate` trait gives you the complete [DE-9IM](https://en.wikipedia.org/wiki/DE-9IM) output, so you can ask questions about how exactly two shapes interact.
